include/__generator.hpp

```diff
@@ -451,18 +464,20 @@ struct __generator_promise_base
         return std::move(__g).get();
     }
 
-    template <std::ranges::range _Rng>
-    __yield_sequence_awaiter<generator<_Ref>> yield_value(std::ranges::elements_of<_Rng> __x) {
+    template <std::ranges::range _Rng, typename _Allocator>
+    __yield_sequence_awaiter<generator<_Ref, std::ranges::range_value_t<_Rng>, _Allocator>>
+    __yield_value(std::ranges::elements_of<_Rng, _Allocator> && __x) {
         // TODO: Consider propagating parent coroutine's allocator to child generator
         // coroutine here.
-        return yield_value(std::ranges::elements_of([](_Rng&& __rng) -> generator<_Ref> {
+        // TODO: Should the promise type be templated on value to reduce template instantiations?
+        return [](auto && __rng, allocator_arg_t, _Allocator alloc) -> generator<_Ref, std::ranges::range_value_t<_Rng>, _Allocator> {
```

Comment:
Note that _Allocator here will potentially be an lvalue reference in the case that an lvalue was passed to `std::ranges::elements_of()`. Is that intentional, or do we want this to always take a copy of the allocator by-value?

I guess passing by reference should be ok here since we aren't using the allocator internally in the coroutine.
The fact that the argument is a reference would mean that the parameter has a trivial destructor which might make it a more obvious candidate for leaving out of the coroutine-frame if it's not used in the coroutine body. But then why not always pass it by reference if that's safe?

Reply:
Yep, That was a mistake on my part